astropy/__init__.py

```diff
@@ -241,10 +242,10 @@ def _rollback_import(message):
         from .utils import _compiler
     except ImportError:
         if is_astropy_source_dir:
-            log.warn('You appear to be trying to import astropy from '
+            warn('You appear to be trying to import astropy from '
                      'within a source checkout without building the '
                      'extension modules first.  Attempting to (re)build '
-                     'extension modules:')
+                     'extension modules:', AstropyWarning)
```

Comment:
This is fine, though it turns out it's not actually necessary since `log.warn` will forward through the warnings interface anyways.